src/agent/src/rpc.rs

```diff
@@ -1218,7 +1218,9 @@ impl agent_ttrpc::AgentService for AgentService {
             s.running = true;
 
             if !req.guest_hook_path.is_empty() {
-                let _ = s.add_hooks(&req.guest_hook_path).map_err(|e| {
+                let timeout: Option<i32> = Some(req.guest_hook_timeout);
```

Comment:
As I mentioned in the earlier comment, it should accept zero timeouts in case admins aren't sure about how long the hook will execute. It seems that the configuration comment is changed but the actual logic is not implemented yet?

Reply:
Previously the default timeout is 10 sec. I would suggest the 0 sec's semantic could be blocking. WDYT?

Reply:
sgtm!

Reply:
\cc @soulseen , PTAL.

Reply:
@Ji-Xinyou @bergwolf agree and updated, please take a review. thanks!

Reply:
> 1. The guest hook is a kata-defined hook. It is not an OCI hook.
> 2. We should allow zero timeout value for those who aren't sure about how long their hooks are expected to execute.
> 
> I'd suggest something like this:
> 
> ```
> Guest hook timeout is the number of seconds to wait for a hook execution before aborting it.
> It must be a non-negative integer. Zero means no timeout. The default value is 10 seconds.
> ```

@Ji-Xinyou @bergwolf I think the `guest hook` here is confusing. actually, it will be passed to OCI Hook struct and the `prestart hook` will be executed after the guest VM start and before the container start.